From ab79f4369853fdd122a78a30901e5e6eea90e1ee Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Mon, 3 Dec 2018 20:59:52 +0530
Subject: [PATCH 0846/1051] net: octeontx2: Add __weak function symbols for
 mbox handlers

In case where mbox handlers are outside kernel driver,
having __weak function symbols for mbox handlers that are no op,
enables us to add mbox ID without the hassle of
defining mbox handler in kernel driver.

Change-Id: I9a078116c74ecd3fc2b96a69c381ba1448e456ce
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Signed-off-by: Linu Cherian <linu.cherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1753
Reviewed-by: Prakash Brahmajyosyula <Brahmajyosyula.Prakash@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/otx2_common.c    | 13 +++++++++++++
 .../net/ethernet/marvell/octeontx2/otx2_common.h    |  9 +++++++++
 drivers/net/ethernet/marvell/octeontx2/otx2_pf.c    |  6 +++---
 3 files changed, 25 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_common.c b/drivers/net/ethernet/marvell/octeontx2/otx2_common.c
index fab7163c34fb..a999e6bc26fd 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_common.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_common.c
@@ -1193,3 +1193,16 @@ void otx2_free_cints(struct otx2_nic *pfvf, int n)
 		free_irq(vector, &qset->napi[qidx]);
 	}
 }
+
+#define M(_name, _id, _fn_name, _req_type, _rsp_type)			\
+int __weak								\
+otx2_mbox_up_handler_ ## _fn_name(struct otx2_nic *pfvf,		\
+				struct _req_type *req,			\
+				struct _rsp_type *rsp)			\
+{									\
+	/* Nothing to do here */					\
+	return 0;							\
+}
+
+MBOX_UP_CGX_MESSAGES
+#undef M
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_common.h b/drivers/net/ethernet/marvell/octeontx2/otx2_common.h
index af2ad4655bdd..cb07495ae9d4 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_common.h
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_common.h
@@ -318,6 +318,15 @@ static struct _req_type __maybe_unused					\
 MBOX_MESSAGES
 #undef M
 
+#define M(_name, _id, _fn_name, _req_type, _rsp_type)			\
+int									\
+otx2_mbox_up_handler_ ## _fn_name(struct otx2_nic *pfvf,		\
+				struct _req_type *req,			\
+				struct _rsp_type *rsp);			\
+
+MBOX_UP_CGX_MESSAGES
+#undef M
+
 #define	RVU_PFVF_PF_SHIFT	10
 #define	RVU_PFVF_PF_MASK	0x3F
 #define	RVU_PFVF_FUNC_SHIFT	0
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
index 6262ef7c89a0..dad9e7efadf9 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
@@ -115,9 +115,9 @@ static void otx2_pfaf_mbox_handler(struct work_struct *work)
 	otx2_write64(af_mbox->pfvf, RVU_PF_INT, BIT_ULL(0));
 }
 
-static int otx2_mbox_up_handler_cgx_link_event(struct otx2_nic *pf,
-					       struct cgx_link_info_msg *msg,
-					       struct msg_rsp *rsp)
+int otx2_mbox_up_handler_cgx_link_event(struct otx2_nic *pf,
+					struct cgx_link_info_msg *msg,
+					struct msg_rsp *rsp)
 {
 	struct cgx_link_user_info *linfo = &msg->link_info;
 	struct net_device *netdev = pf->netdev;
-- 
2.17.1

